lightning/src/ln/channelmanager.rs

```diff
@@ -1765,6 +1765,11 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 						msgs::DecodeError::UnknownRequiredFeature|
 						msgs::DecodeError::InvalidValue|
 						msgs::DecodeError::ShortRead => 0x4000 | 22, // invalid_onion_payload
+						msgs::DecodeError::UnknownTLV(tlv_type) => {
+							let tlv_type = BigSize(tlv_type);
+							let mut v = Vec::with_capacity(tlv_type.serialized_length());
+							return_err!("Unable to decode our hop data", 0x4000 | 22, if let Ok(_) = tlv_type.write(&mut v) { &v[..] } else { &[0;0] });
```

Comment:
Do we need to bother here? Spec says informing the sender of which type was unknown is optional and supporting it means some amount of effort on the TLV-decode end. We can add it if you want, but supporting sending a new onion error type seems unrelated to this PR (which is about handling a new onion error type).

Reply:
Yeah then I'll rather leave it for another PR and just focus on the handling :)